src/ui/components/LedgerViewer.js

```diff
@@ -254,6 +299,69 @@ const LedgerEventRow = React.memo(
               />
             </Box>
           ));
+        case "TRANSFER_GRAIN":
+          let sender;
+          let recipient;
+          const amount = G.formatAndTrim(action.amount);
+          try {
+            const senderAccount = ledger.account(action.from);
+            sender = (
+              <>
+                <IdentityDetails
+                  id={senderAccount.identity.id}
+                  name={senderAccount.identity.name}
+                />
+                <Chip
+                  className={classes.chip}
+                  label={senderAccount.identity.subtype}
+                  size="small"
+                />
+              </>
+            );
+          } catch (e) {
+            console.warn(
+              "Unable to find account for sender in action: ",
+              action
+            );
+            sender = (
+              <IdentityDetails id={action.from} name="[Unknown Account]" />
+            );
+          }
+
+          try {
+            const recipientAccount = ledger.account(action.to);
+            recipient = (
+              <>
+                <IdentityDetails
+                  id={recipientAccount.identity.id}
+                  name={recipientAccount.identity.name}
+                />
+                <Chip
+                  className={classes.chip}
+                  label={recipientAccount.identity.subtype}
+                  size="small"
+                />
+              </>
+            );
+          } catch (e) {
+            console.warn(
+              "Unable to find account for recipient in action: ",
+              action
+            );
+            recipient = (
+              <IdentityDetails id={action.to} name="[Unknown Account]" />
+            );
+          }
+
+          return (
+            <>
+              {sender}
+              &nbsp; &rArr; &nbsp;
+              {amount}
+              &nbsp; &rArr; &nbsp;
+              {recipient}
+            </>
+          );
```

Comment:
Should extract this logic into its own component since there's a decent amount of logic here.

Reply:
Done